js/packages/openinference-instrumentation-langchain/src/types.ts

```diff
@@ -0,0 +1,35 @@
+import { SemanticConventions } from "@arizeai/openinference-semantic-conventions";
+
+type LLMMessageToolCall = {
+  [SemanticConventions.TOOL_CALL_FUNCTION_NAME]?: string;
+  [SemanticConventions.TOOL_CALL_FUNCTION_ARGUMENTS_JSON]?: string;
+};
+export type LLMMessage = {
+  [SemanticConventions.MESSAGE_ROLE]?: string;
+  [SemanticConventions.MESSAGE_CONTENT]?: string;
+  [SemanticConventions.MESSAGE_FUNCTION_CALL_NAME]?: string;
+  [SemanticConventions.MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON]?: string;
+  [SemanticConventions.MESSAGE_TOOL_CALLS]?: LLMMessageToolCall[];
+};
+
+export type LLMMessageFunctionCall = Pick<
+  LLMMessage,
+  "message.function_call_name" | "message.function_call_arguments_json"
+>;
+
+export type LLMMessageToolCalls = Pick<LLMMessage, "message.tool_calls">;
```

Comment:
curious why the Pick approach was taken here. Wouldn't it be better to do a Union type of sub-types.

Reply:
I think it was just order of operations, i wrote the top type first then extracted from it, might be more clear to do a union of the bjects though so can do that